### PR TITLE
EC-33 First commit for Linkedin Learning connector

### PR DESCRIPTION
**Pull request by NileshN**
@amardaxini @yogendra689 @akshayEdCast Kindly review Linkedin Learning content collector. 
Thanks.
Copy: @darshan-patel

### CODE REVIEW

File: app/service/linkedin_learning_integration.rb [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+    AppConfig.integrations['linkedin_learning']['ecl_client_id']
+  end
+
+  def self.ecl_token
+    AppConfig.integrations['linkedin_learning']['ecl_token']
+  end
+
+  def courses_url(current_page, per_page)
+    start = (current_page.to_i * per_page) + 1
+    param = { "q": "localeAndType",
```
Comment (pallavsharma) [inline app/service/linkedin_learning_integration.rb]: hi can we follow safari integration here,
keep params out of url like `json_request(url, method, params:{}, headers:{}, basic_auth:{}, bearer:nil, body: nil)`

File: app/service/linkedin_learning_integration.rb [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+    deep_link_url = ''
+    # aiccLaunch: the launch URL of the learning asset that can be used to initiate AICC tracking in an AICC-compliant system.
+    # aiccLaunch is the URL of LinkedIn Learning content, a URL that includes an identifier for a specific organization
+    # webLaunch : the launch URL of the learning asset in the LinkedIn Learning web application.
+    if details.try(:[], "urls").try(:[], "aiccLaunch")
+      deep_link_url =  details['urls']['aiccLaunch']
+    elsif details.try(:[], "urls").try(:[], "webLaunch")
+      deep_link_url =  details['urls']['webLaunch']
+    end
+
```
[the remark this answers is not in the document]
Comment (NileshN) [reply app/service/linkedin_learning_integration.rb]: @yogendra689  .try will handle `nil`

File: app/service/linkedin_learning_integration.rb (hunk 1 above)
Comment (NileshN) [reply app/service/linkedin_learning_integration.rb]: @pallavsharma Updated code. please check [commit](https://github.com/Course-Master/collector_app/pull/4/commits/4845b7bfe66545580a3b4c8c0ba12cc8273e260c)

File: app/service/linkedin_learning_integration.rb [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+              "expandDepth": "1",
+              "includeRetired": "false",
+              "start": start,
+              "count": per_page }
+    "#{LINKEDIN_LEARNING_BASE_URL}/learningAssets?#{param.to_query}"
+  end
+
+  def get_content(options={})
+    begin
+      per_page = 20
```
[the remark this answers is not in the document]
Comment (NileshN) [reply app/service/linkedin_learning_integration.rb]: @yogendra689 Updated code. please check [commit](https://github.com/Course-Master/collector_app/pull/4/commits/4845b7bfe66545580a3b4c8c0ba12cc8273e260c)

File: app/service/linkedin_learning_integration.rb [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+      courses_paginated_url = courses_url(options[:page], per_page)
+      data = json_request(courses_paginated_url, :get, params:{}, headers: { 'Authorization' => "Bearer #{get_access_token}", 'referer' => 'urn:li:partner:edcast' })
```
[the remark this answers is not in the document]
Comment (NileshN) [reply app/service/linkedin_learning_integration.rb]: @yogendra689 Updated code. please check [commit](https://github.com/Course-Master/collector_app/pull/4/commits/4845b7bfe66545580a3b4c8c0ba12cc8273e260c)

File: app/service/linkedin_learning_integration.rb (hunk 2 above)
[the remark this answers is not in the document]
Comment (amardaxini) [reply app/service/linkedin_learning_integration.rb]: try is ideally is not good practice, @yogendra689  has made a perfect suggestion,
Moreover multiple try is not readable either

[the remark this answers is not in the document]
Comment (NileshN) [reply app/service/linkedin_learning_integration.rb]: @amardaxini @yogendra689 Please check [commit](https://github.com/Course-Master/collector_app/commit/c89cbd2c564b26620439f1ca3d90fb57a0b26219)

**Pull request merged**